pkg/lib/authn/authenticator/password/provider.go

```diff
@@ -124,12 +125,17 @@ func (p *Provider) Authenticate(a *authenticator.Password, password string) (req
 		}
 	}
 
-	if notAllowedErr := p.PasswordChecker.ValidateCurrentPassword(password); notAllowedErr != nil {
+	if validateErr := p.PasswordChecker.ValidateCurrentPassword(a.UserID, password); validateErr != nil {
 		if p.Config.ForceChange != nil && *p.Config.ForceChange {
 			requireUpdate = true
 		}
 	}
 
+	if expiryErr := p.Expiry.Validate(password, a.ID); expiryErr != nil {
```

Comment:
It should take `a` instead of `password`. We just need to check the password being used is expired, not all password the user has.